core/src/main/java/com/epam/esm/dao/impl/GiftDaoImpl.java

```diff
@@ -0,0 +1,282 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftDao;
+import com.epam.esm.dao.exception.GiftNotFoundException;
+import com.epam.esm.dao.model.QueryArgModel;
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+import com.epam.esm.model.entity.TagEntity;
+import com.epam.esm.util.SearchConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class GiftDaoImpl implements GiftDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    private static final String FIND_ALL_GIFTS_QUERY = "select * from gift_certificate";
+    private static final String FIND_GIFT_BY_ID_QUERY = "select * from gift_certificate where id = ?";
+
+
+
+    private static final String BY_NAME = " where name like ?";
+    private static final String BY_DESCRIPTION = " where description like ?";
+    private static final String BY_NAME_AND_DESCRIPTION = " where name like ? and description like ?";
+
+
+    private static final String SELECT_GIFT_BY_NAME = FIND_ALL_GIFTS_QUERY + BY_NAME;
```

Comment:
I would extract all queries into separate class, e.g. GiftDaoQueries. This class can be package-private 
For now there is too many constants in one class